src/Client/Updater.php

```diff
@@ -427,9 +437,12 @@ private function searchDelegatedRolesForTarget(TargetsMetadata $targetsMetadata,
         foreach ($targetsMetadata->getDelegatedKeys() as $keyId => $delegatedKey) {
             $this->signatureVerifier->addKey($keyId, $delegatedKey);
         }
+        if (!isset($this->delegatedRoles[$targetsMetadata])) {
+            $this->delegatedRoles[$targetsMetadata] = $targetsMetadata->getDelegatedRoles();
```

Comment:
We can use null-coalescing assignment for this to remove the `if` check.

```suggestion
            $this->delegatedRoles[$targetsMetadata] ??= $targetsMetadata->getDelegatedRoles();
```